src/ucp/core/ucp_request.inl

```diff
@@ -101,6 +102,7 @@
             } \
         } else { \
             __req = ((ucp_request_t*)(_param)->request) - 1; \
+            ucp_request_super_reset(__req); \
```

Comment:
reqeust alloc macros become too bloated
need to add inline functions to take away code from macros:
1. ucp_request_reset_mpool: ucp_request_super_reset, id_check, VALGRIND_MAKE_MEM_DEFINED
2. ucp_request_reset_external: ucp_request_super_reset, id_reset

Reply:
done,
but I guess it is better to have `ucp_request_reset()` which is common for both cases and use it directly for resetting external requests and from `ucp_request_reset_mpool()` when allocating from mpool